index/server/pkg/server/endpoint.go

```diff
@@ -543,6 +543,12 @@ func buildIndexAPIResponse(c *gin.Context, indexType string, wantV1Index bool, p
 	if params.Deprecated != nil {
 		util.FilterDevfileDeprecated(&index, *params.Deprecated, wantV1Index)
 	}
+	if index == nil || len(index) == 0 {
+		c.JSON(http.StatusOK, gin.H{
```

Comment:
```suggestion
		c.JSON(http.StatusInternalServerError, gin.H{
```

I'm wondering if this should be a [500](https://developer.mozilla.org/en-US/docs/Web/HTTP/Status/500) instead of a 200 as it not having the samples is technically an error and is referenced as an error in the [original issue](https://developer.mozilla.org/en-US/docs/Web/HTTP/Status/500) 


https://go.dev/src/net/http/status.go

wdyt @michael-valdron

Reply:
@Jdubrick Maybe [404](https://developer.mozilla.org/en-US/docs/Web/HTTP/Status/404) would be more appropriate error code here, since no stacks or samples can be found.

Reply:
I was thinking `500` as it would be the servers fault no samples were loaded, no? I am also okay with `404` though

Reply:
> I was thinking `500` as it would be the servers fault no samples were loaded, no? I am also okay with `404` though

That is a good point, but in this case the control logic that provides this response does not guarantee that the stacks and samples are not loaded due to error versus just having an empty registry.

Reply:
I'm okay with `404` in that case. 

@Horiodino for reference that is `http.StatusNotFound`

Reply:
IMO, in a world where we're working towards more dynamic devfile registries, I'd argue that a 200 code is fine here.